blogposts/2020/optimizing-a-code-intelligence-indexer.md

```diff
@@ -0,0 +1,196 @@
+---
+title: "Optimizing a code intelligence indexer"
+author: Eric Fritz
+authorUrl: https://eric-fritz.com
+publishDate: 2020-08-17T08:00-07:00
+tags: [blog]
+slug: optimizing-a-code-intel-indexer
+heroImage: https://sourcegraphstatic.com/blog/lsif-go/lsif-go-improvements.png
+published: true
+---
+
+We've finally tagged the [one-point-oh release](https://github.com/sourcegraph/lsif-go/releases/tag/v1.0.0) of [lsif-go](https://github.com/sourcegraph/lsif-go), boasting impressive speed improvements that bring intelligence to your Go code faster than ever before.
```

Comment:
My interpretation is that this post intends to communicate 2 things:

- "Sourcegraph code intelligence is fast on big repos" to people with big repos
- "We are thoughtful, smart devs" to other devs (broad audience)

Is that right? I think those are great things to communicate. I would recommend changing the intro to remove the mention of tagging the release and other details that are irrelevant (IMO) to these points, and just focus on nailing the 2 points above. Something like this for the intro:


```
On the [code intelligence team](handbook-link) at Sourcegraph, we want you to always have precise and up-to-date hovers, definitions, and references when you're browsing and reviewing unfamiliar code, even on very large monorepos.

We call our key (internal) metric **time to code intelligence** (TTCI) [also link to this in the handbook, so you bolster its credibility as a metric], which means the time it takes to: (phrasing TBD)

1. Push a new commit to a branch on your repository
1. (List all other steps that usually occur, such as CI running and uploading)
1. Code intelligence is ready for use on Sourcegraph

(Some sentence about how companies with all kinds of codebases and repo sizes use Sourcegraph and how that's given us a lot of input.) We've made this much faster lately. Here are some specific performance improvements we made (and how we identified them as high-priority).

```

Notes:

- Precisely defining the `time to intelligence` metric will help establish our claim to being `thoughtful, smart devs`. An imprecise definition will cause us to lose credibility.
- Mentioning specific versions makes the blog post seem ephemeral. The contents and the message are timeless, and I don't want to ruin that positive characteristic.

Reply:
Noted and integrating.

We currently don't have a formal definition for TTI in the handbook (but we do have it marked as one of formal team goals that @aidaeology's taking point on https://about.sourcegraph.com/handbook/engineering/code-intelligence).